rosbag2_py/src/rosbag2_py/_transport.cpp

```diff
@@ -127,14 +128,43 @@ typedef OptionsWrapper<rosbag2_transport::RecordOptions> RecordOptions;
 namespace rosbag2_py
 {
 
+
+class Arguments {
+public:
+  Arguments(const std::vector<std::string>& args) : arguments_(args) {
+    std::for_each(arguments_.begin(), arguments_.end(),
+        [this](const std::string& arg){
+          pointers_.push_back(const_cast<char*>(arg.c_str()));
+        }
+    );
+    pointers_.push_back(NULL);
+  }
+
+  char ** argv()
+  {
+    return arguments_.empty() ? nullptr : pointers_.data();
+  }
+
+  int argc() const
+  {
+    return arguments_.size();
+  }
+
+private:
+  std::vector<std::string> arguments_;
+  std::vector<char*> pointers_;
+};
```

Comment:
Does it make more sense to move this into `rosbag2_cpp`, it already has dependency on `rosbag2_cpp`?

Reply:
@fujitatomoya As regards
> Does it make more sense to move this into rosbag2_cpp, it already has dependency on rosbag2_cpp?

Well, in theory, we can. It likely will give us the ability to reuse the same class in the `rosbag2_py::Player` and `rosbag2_py::Recorder` classes.
However, in this case we likely will need to make API/ABI breaking changes on the `rosbag2_cpp` layer.  
But we want to backport this PR down to the Humble if possible. And with the current design/approach it seems it will be doable since there are no API/ABI breaking changes so far.
Another point to consider is that moving this parameter's wrapper class to the `rosbag2_cpp` is too deep in the rosbag2 hierarchy and we unlikely ever will be using this newly created class on the `rosbag2_transport` or other really cpp layers. It is only useful on the `rosbag2_py` layer.
BTW, with the current `Arguments` class declaration inside `_transport.cpp`, nothing stops us from using it for the `rosbag2_py::Recorder` class as well.

Reply:
Left in the same file, but moved to unnamed namespace